doc/book/cookbook/setting-locale-depending-routing-parameter.md

```diff
@@ -0,0 +1,303 @@
+# How can I setup the locale depending on a routing parameter?
+
+It is a common task to have an localized web application where the setting of
+the locale (and therefor the language) depends on the routing. There are other 
+ways of setting a language like using the session or a specialized sub-domain.
+In this recipe we will concentrate on using the a routing parameter.
+
+## Setting up the route ##
+
+If you want to set the locale depending on an routing parameter, you first have
+to add a language parameter to your routes. In this example we use the `lang` 
```

Comment:
But how would you handle redirects within a middleware? Currently I do this:

```
$routeParams = [
    'id'   => $id,
    'lang' => $request->getAttribute('lang'),
];

return new RedirectResponse(
    $this->router->generateUri('pizza-show', $routeParams)
);
```

That could be changed to 

```
$routeParams = [
    'id'   => $id,
    'lang' => Locale::getDefault(),
];

return new RedirectResponse(
    $this->router->generateUri('pizza-show', $routeParams)
);
```

But then again, in most projects I don't want to have `de_DE` or `en_US` in my URLs, I would preper `de` and `en`. So I could use `Locale::getPrimaryLanguage(Locale::getDefault())` to achieve that.

Sounds reasonable to me. Though I need to think about restructuring the cook book for both ideas. Personally I like the idea of the route result observer.